# SQLite does not support multiple DDL statements in a single transaction. As a result, Drizzle
# inserts a comment (`--> statement-breakpoint`) between each DDL statement in the migration file
# so that it can split the contents of the migration file and run each statement one-by-one. See
# https://orm.drizzle.team/docs/drizzle-config-file#breakpoints.
#
# The linter inserts a space between the `--` and the `>` in the comment which causes migrations
# to fail. This patch modifies Drizzle's migrator to look for the comment with a space between the
# `--` and the `>` instead of the original comment.
#
# Related: https://github.com/drizzle-team/drizzle-orm/issues/3218.

diff --git a/node_modules/drizzle-orm/migrator.cjs b/node_modules/drizzle-orm/migrator.cjs
index c239e1d..2708e9a 100644
--- a/node_modules/drizzle-orm/migrator.cjs
+++ b/node_modules/drizzle-orm/migrator.cjs
@@ -46,7 +46,7 @@ function readMigrationFiles(config) {
     const migrationPath = `${migrationFolderTo}/${journalEntry.tag}.sql`;
     try {
       const query = import_node_fs.default.readFileSync(`${migrationFolderTo}/${journalEntry.tag}.sql`).toString();
-      const result = query.split("--> statement-breakpoint").map((it) => {
+      const result = query.split("-- > statement-breakpoint").map((it) => {
         return it;
       });
       migrationQueries.push({
diff --git a/node_modules/drizzle-orm/migrator.js b/node_modules/drizzle-orm/migrator.js
index 3a3f0b4..3aabe27 100644
--- a/node_modules/drizzle-orm/migrator.js
+++ b/node_modules/drizzle-orm/migrator.js
@@ -13,7 +13,7 @@ function readMigrationFiles(config) {
     const migrationPath = `${migrationFolderTo}/${journalEntry.tag}.sql`;
     try {
       const query = fs.readFileSync(`${migrationFolderTo}/${journalEntry.tag}.sql`).toString();
-      const result = query.split("--> statement-breakpoint").map((it) => {
+      const result = query.split("-- > statement-breakpoint").map((it) => {
         return it;
       });
       migrationQueries.push({
diff --git a/node_modules/drizzle-orm/op-sqlite/migrator.cjs b/node_modules/drizzle-orm/op-sqlite/migrator.cjs
index 5c645c7..508cce9 100644
--- a/node_modules/drizzle-orm/op-sqlite/migrator.cjs
+++ b/node_modules/drizzle-orm/op-sqlite/migrator.cjs
@@ -31,7 +31,7 @@ async function readMigrationFiles({ journal, migrations }) {
       throw new Error(`Missing migration: ${journalEntry.tag}`);
     }
     try {
-      const result = query.split("--> statement-breakpoint").map((it) => {
+      const result = query.split("-- > statement-breakpoint").map((it) => {
         return it;
       });
       migrationQueries.push({
diff --git a/node_modules/drizzle-orm/op-sqlite/migrator.js b/node_modules/drizzle-orm/op-sqlite/migrator.js
index 28f71a9..8bf4eb6 100644
--- a/node_modules/drizzle-orm/op-sqlite/migrator.js
+++ b/node_modules/drizzle-orm/op-sqlite/migrator.js
@@ -7,7 +7,7 @@ async function readMigrationFiles({ journal, migrations }) {
       throw new Error(`Missing migration: ${journalEntry.tag}`);
     }
     try {
-      const result = query.split("--> statement-breakpoint").map((it) => {
+      const result = query.split("-- > statement-breakpoint").map((it) => {
         return it;
       });
       migrationQueries.push({
